packages/flutter_tools/lib/src/intellij/intellij.dart

```diff
@@ -69,17 +69,67 @@ class IntelliJPlugins {
     return _fileSystem.isDirectorySync(packagePath);
   }
 
-  String _readPackageVersion(String packageName) {
-    final String jarPath = packageName.endsWith('.jar')
-        ? _fileSystem.path.join(pluginsPath, packageName)
-        : _fileSystem.path.join(pluginsPath, packageName, 'lib', '$packageName.jar');
-    final File file = _fileSystem.file(jarPath);
-    if (!file.existsSync()) {
-      return null;
+  ArchiveFile _findPluginXml(String packageName) {
+    final List<String> mainJarPathList = <String>[];
+    if (packageName.endsWith('.jar')) {
+      // package exists(checked in _hasPackage
+      mainJarPathList.add(_fileSystem.path.join(pluginsPath, packageName));
+    } else {
+      final String packageLibPath =
+          _fileSystem.path.join(pluginsPath, packageName, 'lib');
+      if (!_fileSystem.isDirectorySync(packageLibPath)) {
+        return null;
+      }
+      // Collect the files with a file suffix of .jar/.zip that contains the plugin.xml file
+      final List<FileSystemEntity> pluginJarPaths = _fileSystem
+          .directory(_fileSystem.path.join(pluginsPath, packageName, 'lib'))
+          .listSync()
```

Comment:
since you only care about `File`s, you can add `whereType<File>()` to filter the iterable down to just them.  Then you can skip the stat check here and make `pluginJarPaths ` a List of files:

```
_fileSystem
    .directory(_fileSystem.path.join(pluginsPath, packageName, 'lib'))
    .listSync()
    .whereType<File>()
    .where((File file) {
     final String fileExt= _fileSystem.path.extension(file.path);
     return fileExt == '.jar' || fileExt == '.zip';
   })
   .toList()
```

Reply:
Thank you. I fixed it like you commented.